x-pack/plugins/lists/common/schemas/common/schemas.ts

```diff
@@ -276,11 +276,12 @@ export type CursorOrUndefined = t.TypeOf<typeof cursorOrUndefined>;
 export const namespace_type = DefaultNamespace;
 
 export const operatorIncluded = t.keyof({ included: null });
-export const operator = t.keyof({ excluded: null, included: null });
+export const operator = t.keyof({ excluded: null, included: null, wildcard_caseless: null });
 export type Operator = t.TypeOf<typeof operator>;
 export enum OperatorEnum {
   INCLUDED = 'included',
   EXCLUDED = 'excluded',
+  WILDCARD_CASELESS = 'wildcard_caseless',
```

Comment:
`included` and `excluded` are used as boolean negation operators, allowing us to choose to allowlist documents that match a query or allowlist documents that do not match the query. `wildcard_caseless` is a new type of entry so it belongs in the `OperatorTypeEnum` below.

Reply:
I see. I had to add `wildcard_caseless` as a new value to `operator` as well since a function in trusted_apps ends up using `operator` as its type. I could use some help in decoupling these types if possible.

Reply:
You'll want to add another entry type similar to https://github.com/elastic/kibana/blob/master/x-pack/plugins/lists/common/schemas/types/entry_match.ts, except with `type: t.keyof({ wildcard_caseless: null }),`, then add that new entry type to the `entriesArray` union (https://github.com/elastic/kibana/blob/master/x-pack/plugins/lists/common/schemas/types/entries.ts#L20).

I'm happy to zoom/pair on it if you want as well.

Reply:
Hi @marshallmain I'm helping out Ash with this and following your feedback - thanks for that. We (trusted apps) seem to be using `EntriesArray`, but now that I'm looking at the types closer in lists, I see there is also a `EndpointEntriesArray` - is that one more for exceptions that apply to endpoint? if so, I'm thinking that is really the one we probably should have been using all along.

Reply:
Yeah `EndpointEntriesArray` is the schema for exceptions that the endpoint can support, however it's used in such a way that essentially `EndpointEntriesArray extends EntriesArray`. `EndpointEntriesArray` is used as additional validation after validating with `EntriesArray` when adding exceptions to the special endpoint exception list. For trusted apps, you could implement a similar subset `TrustedAppsEntriesArray` specifying the subset of entry types that are valid for trusted apps, but since it's not the same subset as `EndpointEntriesArray` it should be defined separately. It appears there is already other logic for trusted apps that checks if the entry is valid for trusted apps though.

When you call `exceptionsListClient.createExceptionListItem` the `entries` passed in must be of type `EntriesArray`, so all possible entry types need to be included there. This also ensures that the ExceptionListClient return types are correct.

Reply:
yeah, I see that now. Adding a new type to the `EntriesArray` has a broad impact through out exceptions UI. I'm going to look to see if we can get around that by forcing something in trusted apps (maybe a cast somewhere)

Reply:
@ashokaditya looks like there is no good way to avoid having to change `EntriesArray` to include a new type. We could cast our own type into `EntriesArray` when using the mappers or exception client, but doing so could introduce issues in the future and is probably not the best way to go about this. 

here is the type I crated locally (under `x-pack/plugins/lists/common/schemas/types/entry_match_wildcard_caseless.ts`):

```typescript
import * as t from 'io-ts';

import { NonEmptyString } from '../../shared_imports';

export const entriesMatchWildcardCaseless = t.exact(
  t.type({
    field: NonEmptyString,
    operator: t.keyof({ wildcard_caseless: null }),
    type: t.keyof({ match: null }),
    value: NonEmptyString,
  })
);
export type EntriesMatchWildcardCaseless = t.TypeOf<typeof entriesMatchWildcardCaseless>;
```

Reply:
The endpoint [wildcard tests](https://github.com/elastic/endpoint-dev/blob/master/Elastic/Libraries/FilterLib/Tests/filter_data/test-wildcard-10/allow_list.json) have entries structured like 
```
{
  "operator": "included",
  "field": "process.parent.executable",
  "type": "wildcard_caseless",
  "value": "/*???/launchd"
}
```
so to match that and avoid introducing the new operator type specifically for `wildcard_caseless` the io-ts type would be
```typescript
import * as t from 'io-ts';

import { NonEmptyString } from '../../shared_imports';
import { operator } from '../common/schemas';

export const entriesMatchWildcardCaseless = t.exact(
  t.type({
    field: NonEmptyString,
    operator,
    type: t.keyof({ wildcard_caseless: null }),
    value: NonEmptyString,
  })
);
export type EntriesMatchWildcardCaseless = t.TypeOf<typeof entriesMatchWildcardCaseless>;
```
The `type` field is used as the discriminant so we don't want to have multiple types in the union with the same key in `type`.

Reply:
Endpoint also looks like it supports both `wildcard_cased` and `wildcard_caseless`, but the requirements for trusted apps specify caseless only. For the `match` entry type, we use context like the `field` in the entry and the OS of the endpoint during the Kibana exception entry -> Endpoint exception entry translation process to decide if it should be translated to a `_cased` or `_caseless` entry. I would suggest following the same pattern here and applying the `_caseless` suffix during the translation process to keep the entry type generic.

Reply:
Ohhh. So - I completely missed that. Here we were trying to add a new operator, when really what we want is a new type 🤦 . Ok - we'll look some more on Monday. Thanks for all your help @marshallmain 🙏